visual-espresso/visual/src/main/java/com/saucelabs/visual/espresso/VisualCheckOptions.java

```diff
@@ -1,17 +1,45 @@
 package com.saucelabs.visual.espresso;
 
+import com.saucelabs.visual.espresso.type.RegionIn;
 import com.saucelabs.visual.espresso.utils.TestMetaInfo;
 
+import java.util.List;
+
 public class VisualCheckOptions {
     private String testName;
     private String suiteName;
+    private List<RegionIn> ignoreRegions;
 
-    public VisualCheckOptions() {
-    }
-
-    public VisualCheckOptions(String testName, String suiteName) {
+    private VisualCheckOptions(String testName, String suiteName, List<RegionIn> ignoreRegions) {
         this.testName = testName;
         this.suiteName = suiteName;
+        this.ignoreRegions = ignoreRegions;
+    }
+
+    public static class Builder {
+        private String testName;
+        private String suiteName;
+        private List<RegionIn> ignoreRegions;
+
+        public Builder withTestName(String testName) {
+            this.testName = testName;
+            return this;
+        }
+
+        public Builder withSuiteName(String suiteName) {
+            this.suiteName = suiteName;
+            return this;
+        }
+
+        public Builder withIgnoreRegions(List<RegionIn> ignoreRegions) {
+            this.ignoreRegions = ignoreRegions;
```

Comment:
idea:

since this is a builder I think, someone can try to pass ignore regions multiple times for different ignore regions based on the domain of the test


.withIgnoreRegions(ignoreRegionsForHeaderTime)
.withIgnoreRegions(ignoreRegionsForFooterContact)
.withIgnoreRegions(allDates)
.withIgnoreRegions(allMockNames)

Maybe we should consider list concatenation instead?